scene-lib/src/scenelib/annotations/el/AClass.java

```diff
@@ -224,4 +251,143 @@ public boolean isEmptyValue(AExpression v) {
+        enums.add(className);
+    }
+
+    /**
+     * Returns the set of enum constants for this class, or null if this is not an enum.
+     *
+     * @return the enum constants, or null if this is not an enum
+     */
+    public /*@Nullable*/ List<VariableElement> getEnumConstants() {
+        if (enumConstants == null) {
+            return null;
+        }
+        return ImmutableList.copyOf(enumConstants);
+    }
+
+    /**
+     * Marks this class as an enum.
+     *
+     * @param enumConstants the list of enum constants for the class
+     */
+    public void setEnumConstants(List<VariableElement> enumConstants) {
+        if (this.enumConstants != null) {
+            throw new Error(String.format(
+                    "setEnumConstants was called multiple times with arguments %s and %s",
+                    this.enumConstants, enumConstants));
+        }
+        this.enumConstants = new ArrayList<>(enumConstants);
+    }
+
+    /**
+     * Get the type of the class, or null if it is unknown. Callers should ensure that either:
+     *
+     * <ul>
+     *   <li>{@link #setTypeElement(TypeElement)} has been called, or
+     *   <li>the return value is checked against null.
+     * </ul>
+     *
+     * @return a type element representing this class
+     */
+    public /*@Nullable*/ TypeElement getTypeElement() {
+        return typeElement;
+    }
+
+    /**
+     * Set the type element representing the class.
+     *
+     * @param typeElement the type element representing the class
+     */
+    public void setTypeElement(TypeElement typeElement) {
+        if (this.typeElement == null) {
+            this.typeElement = typeElement;
+        } else if (!this.typeElement.equals(typeElement)) {
+            throw new Error(String.format(
+                "setTypeElement(%s): type is already %s", typeElement, this.typeElement));
+        }
+    }
+
+    /**
+     * Get all the methods that have been vivified on a class.
+     *
+     * @return a map from method signature (in JVM format) to the object representing the method
+     */
+    public Map<String, AMethod> getMethods() {
+        return ImmutableMap.copyOf(methods);
+    }
+
+    /**
+     * Get all the fields that have been vivified on a class.
+     *
+     * @return a map from field name to the object representing the field
+     */
+    public Map<String, AField> getFields() {
+        return ImmutableMap.copyOf(fields);
+    }
+
+    /**
+     * Get the annotations on the class.
+     *
+     * @return the annotations, directly from scenelib
+     */
+    public Collection<? extends Annotation> getAnnotations() {
+        return tlAnnotationsHere;
+    }
+
+    /**
+     * Update the method's fields, using information from the given ExecutableElement.
+     *
+     * @param methodElt the method
+     * @return an interned AMethod representing the method
+     */
+    public AMethod setFieldsFromMethodElement(ExecutableElement methodElt) {
```

Comment:
The name and/or Javadoc of this method should indicate that the method associated with the passed `MethodElement` is vivified by calling this method.